plugins/modules/files/iso_customize.py

```
@@ -0,0 +1,432 @@
+  register: customize_iso_result
+'''
+
+RETURN = r'''
+src_iso:
+    description: path of source ISO file.
+    returned: on success
+    type: str
+    sample: "/path/to/file.iso"
+dest_iso:
+    description: path of the customized iso file.
+    returned: on success
+    type: str
+    sample: "/path/to/customized.iso"
+
+'''
+
+import os
+import traceback
+import re
+import shutil
+import time
+import copy
+
+PYCDLIB_IMP_ERR = None
+try:
+    import pycdlib
+    HAS_PYCDLIB = True
+except ImportError:
+    PYCDLIB_IMP_ERR = traceback.format_exc()
+    HAS_PYCDLIB = False
+
+from ansible.module_utils.basic import AnsibleModule, missing_required_lib
+from ansible.module_utils.common.text.converters import to_native
+
+MODULE_ISO_CUSTOMIZE = None
+
+
+# Create local temporary direction to store files getting from ISO
+def get_local_tmp_dir():
+    dir_name = time.strftime('%Y-%m-%d-%H-%M-%S', time.localtime(time.time()))
+    tmp_dir = os.path.join('/tmp', dir_name)
+
+    try:
+        os.makedirs(tmp_dir)
+    except OSError as err:
+        msg = "Failed to create folder %s with error %s" % (tmp_dir, to_native(err))
+        MODULE_ISO_CUSTOMIZE.fail_json(msg=msg)
+
+    return tmp_dir
+
+
+def iso_get_file(opened_iso, iso_file, tmp_dir):
+    file_name = os.path.basename(iso_file)
+    file_local = os.path.join(tmp_dir, file_name)
+    if os.path.exists(file_local):
+        return 0, file_local
+
+    try:
+        record = opened_iso.get_record(rr_path=iso_file)
+        if record.is_file():
+            opened_iso.get_file_from_iso(file_local, rr_path=iso_file)
+        else:
+            return -1, "%s is not a file in ISO" % iso_file
+    except Exception as err:
+        msg = "Failed to get iso file %s with error: %s" % (iso_file, to_native(err))
+        MODULE_ISO_CUSTOMIZE.fail_json(msg=msg)
+
+    if not os.path.exists(file_local):
+        return -1, "Failed to get file %s in ISO. Does file exists in ISO ?" % iso_file
```

I **strongly** suggest you raise an `Exception` when something goes wrong, instead of returning numerical error codes.